gcalcli/ics.py

```diff
@@ -98,15 +92,35 @@ def CreateEventFromVOBJ(
 
         # NOTE: Reminders added by GoogleCalendarInterface caller.
 
-        # Can only have an end if we have a start, but not the other
-        # way around apparently...  If there is no end, use the start
+        # Can only have an end if we have a start, but not the other way around
+        # apparently...  If there is no end, use the duration if available, or
+        # the start otherwise.
         if hasattr(ve, 'dtend') and ve.dtend.value:
+            if verbose:
+                print('End..........%s' % ve.dtend.value.isoformat())
+                print('Local End....%s' % localize_datetime(ve.dtend.value))
+
             end = ve.dtend.value.isoformat()
             if isinstance(ve.dtend.value, datetime):
                 event['end'] = {'dateTime': end, 'timeZone': default_tz}
             else:
                 event['end'] = {'date': end}
 
+        elif hasattr(ve, 'duration') and ve.duration.value:
+            if verbose:
+                print('Duration.....%s' % ve.duration.value)
+            end = ve.dtstart.value + ve.duration.value
+            if verbose:
+                print('Calculated End........%s' % end.isoformat())
+                print('Calculated Local End..%s' % localize_datetime(end))
+            # Decide based on dtstart; that's what we base our calculation on.
+            # TODO: correct?
```

Comment:
I had a look at this TODO and it looked pretty obvious that yes duration is always counting from dtstart. Removed the TODO comment.